utils/clarity_extract_handler.py

```
@@ -0,0 +1,541 @@
+    # Convert string representation of list to actual list
+    if isinstance(r_codes, str):
+        try:
+            # Use ast.literal_eval instead of json.loads for Python literal evaluation
+            r_codes_list = ast.literal_eval(r_codes)
+        except (ValueError, SyntaxError):
+            # If parsing fails, assume it's not a valid list
+            return False
+    elif isinstance(r_codes, list):
+        r_codes_list = r_codes
+    else:
+        raise ValueError(f"Unexpected type for R_codes: {type(r_codes)}")
```

Ruff (0.14.0)

Prefer `TypeError` exception for invalid type

(TRY004)